src/commands/utils/uploadToS3.test.ts

```diff
@@ -0,0 +1,157 @@
+import assert from "assert";
+import * as sinon from "sinon";
+import { Logger } from "../../logging";
+import * as notifications from "../../notifications";
+import { uploadFileToS3 } from "./uploadToS3";
+
+describe("uploadFileToS3", () => {
+  let sandbox: sinon.SinonSandbox;
+  let fetchStub: sinon.SinonStub;
+  let loggerErrorStub: sinon.SinonStub;
+  let showErrorNotificationStub: sinon.SinonStub;
+  let mockParams: {
+    file: Blob;
+    presignedUrl: string;
+    contentType: string;
+    uploadFormData: { [key: string]: string };
+  };
+
+  beforeEach(() => {
+    sandbox = sinon.createSandbox();
+
+    fetchStub = sandbox.stub(global, "fetch");
+    loggerErrorStub = sandbox.stub(Logger.prototype, "error");
+    showErrorNotificationStub = sandbox.stub(notifications, "showErrorNotificationWithButtons");
+
+    mockParams = {
+      file: new Blob(["test content"], { type: "application/java-archive" }),
+      presignedUrl: "https://test.s3.amazonaws.com/file.jar",
+      contentType: "application/java-archive",
+      uploadFormData: {
+        key: "test-key",
+        policy: "base64-encoded-policy",
+        "x-amz-algorithm": "AWS4-HMAC-SHA256",
+        "x-amz-credential": "test-credential",
+        "x-amz-date": "20240101T000000Z",
+        "x-amz-signature": "test-signature",
+        "x-amz-security-token": "test-security-token",
+      },
+    };
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it("should return response on successful upload", async () => {
+    const mockResponse = new Response(null, {
+      status: 204,
+      statusText: "No Content",
+      headers: new Headers({
+        etag: '"test-etag"',
+        "x-amz-request-id": "test-request-id",
+      }),
+    });
+    fetchStub.resolves(mockResponse);
+
+    const response = await uploadFileToS3(mockParams);
+
+    // check that the form data is correct
+    const expectedFormDataKeys = [...Object.keys(mockParams.uploadFormData), "file"];
```

Comment:
nit: this `FormData with all expected fields` is probably not needed since I assume the test would fail if formdata was malformed.